src/Microsoft.CodeAnalysis.Testing/Microsoft.CodeAnalysis.Analyzer.Testing/AnalyzerTest`1.cs

```diff
@@ -1265,6 +1272,46 @@ static Solution AddProjectReferences(Solution solution, ProjectId sourceProject,
             }
         }
 
+        protected (string fileName, IEnumerable<string> folders) GetNameAndFoldersFromPath(string path)
+        {
+            if (Path.IsPathRooted(path))
+            {
+                // If the user provides a rooted path as the file name, just use that as-is.
+                return (path, NoFoldersArray);
+            }
+
+            // Normalize to platform path separators for simplicity later on
+            var normalizedPath = path.Replace('\\', Path.DirectorySeparatorChar);
+            var normalizedDefaultPathPrefix = DefaultFilePath.Replace('\\', Path.DirectorySeparatorChar);
```

Comment:
⚠️ `DefaultFilePath` only applies to the top-most `ProjectState`. The `AdditionalProjects` collection contains other projects, each with its own default file path. This method should accept the default file path as an argument, and the caller updated to use the one for the project it is currently processing.